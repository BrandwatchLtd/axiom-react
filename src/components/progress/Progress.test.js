import React from 'react';
import renderer from 'react-test-renderer';
import { Progress } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <Progress { ...props } />
  );
}

function matchSnapshot(props = {}) {
  const component = getComponent(props);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}

describe('Progress', () => {
  it('renders with defaultProps', () => {
    matchSnapshot();
  });

  it('renders an infinite progress indicator', () => {
    matchSnapshot();
  });

  it('renders a finite progress indicator', () => {
    matchSnapshot({ percent: 50 });
  });

  it('renders a success indicator', () => {
    matchSnapshot({ percent: 100 });
  });
});
