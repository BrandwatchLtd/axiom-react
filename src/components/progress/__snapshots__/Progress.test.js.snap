// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders a finite progress indicator 1`] = `
<svg
  className="ax-radial-progress ax-radial-progress--small ax-progress-finite"
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-finite__background"
      r={10}
      style={
        Object {
          "strokeWidth": 4,
        }
      }
    />
    <circle
      className="ax-progress-finite__arc"
      r={10}
      style={
        Object {
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 31.41592653589793,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;

exports[`Progress renders a success indicator 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--success"
>
  <svg
    className="ax-icon ax-icon--tick"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path d=\\"M4 7.5l2 2 6-6L13.5 5 6 12.5 2.5 9z\\" fill-rule=\\"nonzero\\" fill=\\"currentColor\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`Progress renders an error indicator 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--error"
>
  <svg
    className="ax-icon ax-icon--warning"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path d=\\"M6.25 2h3.5l-.5 8.25h-2.5L6.25 2zm.375 11c0-.759.617-1.375 1.375-1.375s1.375.616 1.375 1.375c0 .758-.617 1.375-1.375 1.375A1.377 1.377 0 0 1 6.625 13z\\" fill=\\"currentColor\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`Progress renders with defaultProps 1`] = `
<svg
  className="ax-radial-progress ax-radial-progress--small ax-progress-infinite ax-progress-infinite--subtle"
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-infinite__arc"
      r={10}
      style={
        Object {
          "stroke": "url(#ax-progress-infinite__filter--subtle)",
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 15.707963267948966,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;
